### PR TITLE
Fix warnings with GCC 7

### PR DESCRIPTION
**Pull request by quantumsteve**
**Description of work.**

This addresses several warnings present with GCC 7.

**To test:**

Build mantid and AllTests on a machine with GCC 7. Verify that no compiler warnings are issued.

There is no GitHub issue associated with this pull request.

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by peterfpeterson — APPROVED** [review]
No warnings when built on RHEL7 with `devtoolset-7`.

File: buildconfig/CMake/googletest_wconversion.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+diff --git a/googlemock/include/gmock/gmock-matchers.h b/googlemock/include/gmock/gmock-matchers.h
+index 33b37a7..4f52f35 100644
+--- a/googlemock/include/gmock/gmock-matchers.h
++++ b/googlemock/include/gmock/gmock-matchers.h
+@@ -1620,8 +1620,9 @@ class VariadicMatcher {
+   }
+ 
+  private:
++#pragma GCC diagnostic ignored "-Wconversion"
```
Comment (martyngigg) [inline buildconfig/CMake/googletest_wconversion.patch]: Should this be wrapped in an `#ifdef __GNUC__`? I think that would solve the Windows compiler issue.

Comment (quantumsteve) [reply buildconfig/CMake/googletest_wconversion.patch]: Had to be a bit more creative since clang also defines `__GNUC__`

**Comment by peterfpeterson** [comment]
Still ready to merge.

**Pull request merged**